.github/workflows/code-review-checklist.yml

```diff
@@ -0,0 +1,20 @@
+name: Checklists
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+    types:
+      - opened
+      - reopened
+      - edited
```

Comment:
How long will this job take to run? It looks pretty minimal but it also looks like this will trigger a job every time someone clicks a checkbox 😬 

Also, do we want this running on draft PRs?

Reply:
> How long will this job take to run? 

A second or two.

> It looks pretty minimal but it also looks like this will trigger a job every time someone clicks a checkbox

Yes, you are right. This is not a big deal I think, but the fact that it will generate a ton of emails is a little bit annoying 😄. Unfortunately, I haven't found a way to disable notifications for a certain job. If this is a problem, then I think we need to remove this job. cc @felixarntz 

> Also, do we want this running on draft PRs?

Good point, we don't need to run if for draft PRs. Updated it to run only for PRs that are ready for review.

Reply:
> Yes, you are right. This is not a big deal I think, but the fact that it will generate a ton of emails is a little bit annoying 😄

I think it could be problematic as each job will fill up our queue of jobs to run and then run in order they were added which adds some amount of overhead for each separate job, when really only the last one is important.

I like the idea here, but I'm not sure it's worth adding this as an automated check, especially since we don't require all checks to pass before merging anyways. If checking the boxes becomes significant in that a check will fail as a result, engineers will be somewhat incentivized to check them all ahead of time without really paying attention to them. I would say let's keep this managed manually and try to use this as a useful reminder, unless of course @felixarntz would prefer to keep it 😄

Reply:
Yeah, I would prefer to keep this only a checklist for awareness, we don't need to enforce anything at the automation level.